### ISSUE TITLE
Don't `FatalError` on pcap read errors

### ISSUE THREAD

**Issue by ckreibich** [zeek/zeek#3497]
When Zeek processes a pcap, any error that libpcap reports when reading a packet [leads to a `FatalError`](https://github.com/zeek/zeek/blob/master/src/iosource/pcap/Source.cc#L225), so Zeek exits immediately. This is pretty drastic, especially since it seems that often the cause is just that the last packet in the pcap is truncated, so Zeek has technically already done all the work, but now won't finalize it.

I'm thinking we could downgrade the `FatalError` to a regular one, but shut down the packet source at that point:
```
diff --git i/src/iosource/pcap/Source.cc w/src/iosource/pcap/Source.cc
index be01e148c..93e212aae 100644
--- i/src/iosource/pcap/Source.cc
+++ w/src/iosource/pcap/Source.cc
@@ -221,8 +221,10 @@ bool PcapSource::ExtractNextPacket(Packet* pkt) {
             // Error occurred while reading the packet.
             if ( props.is_live )
                 reporter->Error("failed to read a packet from %s: %s", props.path.data(), pcap_geterr(pd));
-            else
-                reporter->FatalError("failed to read a packet from %s: %s", props.path.data(), pcap_geterr(pd));
+            else {
+                reporter->Error("failed to read a packet from %s: %s", props.path.data(), pcap_geterr(pd));
+                Close();
+            }
             return false;
         case 0:
             // Read from live interface timed out (ok).
```
Filtering for the truncation error to handle different cases more selectively seems brittle — we'd have to do pattern-matching on the error message strings generated by libpcap.

This stuff tends to be subtle so I wanted to check with folks here before going straight to a PR.

I'm attaching two [pcaps](https://github.com/zeek/zeek/files/13621435/pcaps.zip) where one is just the other with the last byte cut off, for testing.

**Comment by ckreibich** [comment]
The above changes exit status, btw — probably not a good idea since it changes semantics for tooling that might rely on current semantics.

**Comment by JustinAzoff** [comment]
Wonder if we need something like

```
reporter->DeferredExitError
```
or such.. probably a better name :-) something that logs an error and sets an exit code variable as failed, so that when zeek does exit, it can report the previously ignored error.

The one example I can think of like this is how `tar` will do

```
tar: Exiting with failure status due to previous errors
```

may also need to think about what that exit code would be, since we would still want to indicate that it was a failure of some kind, but not that zeek completely failed to run due to a script syntax error or something.

**Comment by ckreibich** [comment]
> may also need to think about what that exit code would be

Yeah. `tar` is a nice example, it uses 0/1/2 to signify different levels of problems. Or take curl, which has a huge list of exit codes. A Zeeky take on this might be to route various "soft" errors through the script layer so they become configurable, then exit accordingly.
